src/browser/atom-window.coffee

```diff
@@ -73,7 +73,8 @@ class AtomWindow
     @browserWindow.loadUrl @getUrl(loadSettings)
     @browserWindow.focusOnWebView() if @isSpec
 
-    @openLocations(locationsToOpen) unless @isSpecWindow()
+    hasPathToOpen = not (locationsToOpen.length is 1 and not locationsToOpen[0].pathToOpen?)
```

Comment:
Nice work here!

What do you think about moving this down into `openLocations` (maybe as a guard clause), so that if we call it from somewhere else in the future it just works?